pkg/types/jsonpatch.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// JSONPatch represents a json patch for an object
+type JSONPatch struct {
+	// Relative file path within the kustomization for a json patch file.
+	Path string `json:"path" yaml:"path"`
+
+	// Target refers to a Kubernetes object that the json patch will be
+	// applied to. It must refer to a Kubernetes resource under the
+	// purview of this kustomization. Target should use the
+	// raw name of the object (the name specified in its YAML,
+	// before addition of a namePrefix).
+	Target corev1.ObjectReference `json:"target" yaml:"target"`
```

Comment:
I'm not convinced that using `corev1.ObjectReferenc` is better than having a self-defined struct, since there are fields (e.g. UID, ResourceVersion) that we don't need and may also confuse users.

IMO we can either using a self-defined struct or clarify how `ObjectReference` will be used (what fields are used and what fields are ignored).